js/components/LoginPage.jsx

```diff
@@ -10,9 +10,16 @@ var Signup = require('./Signup.jsx');
 var Link = Router.Link;
 
 var LoginPage = React.createClass({
-  mixins: [LoggedInRoute],
+  mixins: [ Router.Navigation ],
 
   getInitialState: function() {
+    // Redirect to the home page if logged in
+    auth.loggedIn(function(loggedIn) {
```

Comment:
There is a problem with transitioning in the callback of auth. I had to fix this by moving the transition into the LoginPage component and using transitionTo. This causes the login page to flash briefly, but because the auth login is now aysnc I don't see how we can avoid it for now.